compiler/src/main/java/com/squareup/anvil/compiler/codegen/dagger/ProvidesMethodFactoryGenerator.kt

```diff
@@ -102,9 +121,81 @@ internal class ProvidesMethodFactoryGenerator : PrivateCodeGenerator() {
         .withJvmSuppressWildcardsIfNeeded(function)
     val returnTypeIsNullable = function.typeReference?.isNullable() ?: false
 
+    return generateFactoryClass(
+        codeGenDir = codeGenDir,
+        clazz = clazz,
+        packageName = packageName,
+        className = className,
+        functionName = functionName,
+        isObject = isObject,
+        parameters = parameters,
+        returnType = returnType,
+        returnTypeIsNullable = returnTypeIsNullable,
+        isProperty = false
+    )
+  }
+
+  @OptIn(ExperimentalStdlibApi::class)
+  private fun generateFactoryClassForProperty(
+    codeGenDir: File,
+    module: ModuleDescriptor,
+    clazz: KtClassOrObject,
+    property: KtProperty
```

Comment:
I wonder if instead of 2 functions that are nearly identical. you could get away with one that accepts a `functionOrProperty: KtCallableDeclaration` and then type checks if it's a `KtProperty`?

Reply:
I was looking at `function.valueParameters` and the decompiled sources don't show that this is an overridden method:
```kotlin
  @NotNull
  public List<KtParameter> getValueParameters() {
  }
```
So I assumed they don't share a common type, but they do. I can probably revert most of the changes, which is quite nice :) Good catch!